src/lua/api-gateway/aws/AwsService.lua

```diff
@@ -201,16 +206,18 @@ function _M:getCredentials()
     return return_obj
 end
 
-function _M:getAuthorizationHeader(http_method, path, uri_args, body)
+function _M:getAuthorizationHeader(http_method, path, uri_args, body, host_override)
     local credentials = self:getCredentials()
+    credentials.aws_endpoint = self:getAWSHost()
     credentials.aws_region = self.aws_region
     credentials.aws_service = self.aws_service
     local awsAuth = AWSV4S:new(credentials)
-    local authorization = awsAuth:getAuthorizationHeader(http_method,
+    local authorization, payload_hash = awsAuth:getAuthorizationHeader(http_method,
         path, -- "/"
         uri_args, -- ngx.req.get_uri_args()
-        body)
-    return authorization, awsAuth, credentials.token
+        body,
+        host_override)
```

Comment:
In S3, the endpoint URL can take two different forms,  one is like https://BUCKETNAME.s3-us-west-1.amazonaws.com/ and the other is like https://s3-us-west-1.amazonaws.com/BUCKETNAME.  The `host_override` argument was intended to handle those.  In addition to it, I wanted to use `getAuthorizationHeader` from outside `AwsService` where the endpoint URL was specified directly in the settings instead of the bucket name.